topics/S3/etcd-migration/src/steps.py

```diff
@@ -0,0 +1,196 @@
+import json
+import time
+
+from src.common import bcolors, global_vars, Automode
+from src.helpers import set_drain_mode, get_migration_status, print_errors_dict, wait_for_drain, send_bash_command, \
+    send_request, continue_to_next_step, continue_to_next_host
+
+
+def step(step_str):
+    def wrap(f):
+        def wrapped_f(*args):
+            print(f"{bcolors.VIOLET}{bcolors.BOLD}Step {global_vars.step_number}: {step_str}{bcolors.ENDC}")
+            global_vars.step_number += 1
+            if global_vars.auto_mode != Automode.AUTO.value:
+                if not continue_to_next_step():
+                    do_revert()
+
+            f(*args)
+        return wrapped_f
+    return wrap
+
+
+@step("Running preliminary checks on current system")
+def system_health_check():
+    output = send_bash_command("weka s3 cluster status -J")
+    hosts_health = json.loads(output)
+    for key, value in hosts_health.items():
```

Comment:
shouldn't we care only about the health of the `shared_data.hosts` hosts?

Reply:
could there be a difference?